warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/ShardedTableMapFile.java

```diff
@@ -68,6 +78,102 @@ public static void setupFile(Configuration conf) throws IOException, URISyntaxEx
         return locations;
     }
     
+    public static void validateShardIdLocations(Configuration conf, String tableName, int daysToVerify, Map<Text,String> shardIdToLocation) {
+        ShardIdFactory shardIdFactory = new ShardIdFactory(conf);
+        // assume true unless proven otherwise
+        boolean isValid = true;
+        for (int daysAgo = 0; daysAgo <= daysToVerify; daysAgo++) {
+            long inMillis = System.currentTimeMillis() - (daysAgo * DateUtils.MILLIS_PER_DAY);
+            String datePrefix = DateHelper.format(inMillis);
+            int expectedNumberOfShards = shardIdFactory.getNumShards(datePrefix);
+            boolean shardsExist = shardsExistForDate(shardIdToLocation, datePrefix, expectedNumberOfShards);
+            if (!shardsExist) {
+                log.warn("Shards for " + datePrefix + " for table " + tableName + " do not exist!");
+                isValid = false;
+                continue;
+            }
+            boolean shardsAreBalanced = shardsAreBalanced(shardIdToLocation, datePrefix);
+            if (!shardsAreBalanced) {
+                log.warn("Shards for " + datePrefix + " for table " + tableName + " are not balanced!");
+                isValid = false;
+            }
+        }
+        if (!isValid) {
+            throw new IllegalStateException("All of today's shards have not been created.  Run bin/ingest/create_shards_since.sh "
+                            + "yyyymmdd, substitute yyyymmdd with today's date.  Ensure that the normal mechanism "
+                            + "for creating shards is running - e.g, crontab");
```

Comment:
It may also be the case that the shards are just not balanced.  I believe this will also run for all sharded tables, like errorShard, so it might be useful to have the table names that had problems in the message.

Reply:
Makes sense, should we fail on first bad table or aggregate all the failures and log everything that is bad?

Reply:
I am fine with the former but I like how you logged each that was wrong and think it would be helpful to include that in the exception message too.  It might seem like overkill, and maybe it is, given that you're writing to the log, but I'm hoping to lower the bar on identifying the solution for folks that aren't as familiar with the system.